Added UTM dropdowns to Analytics Growth pages

9larsons opened the pull request:
ref https://linear.app/ghost/issue/PROD-2557/
ref https://linear.app/ghost/issue/PROD-2559/
- added campaign dropdown to stats and posts Growth views

This PR adds the UTM dropdown to the Growth pages of the Posts and Stats (Analytics) apps. This is currently behind the `utmTracking` flag and wired up to the `/stats/utm-growth` endpoint.

9larsons committed:
Added stats/utm-growth api hook+type

9larsons committed:
Added UTM campaign filtering to Growth analytics

- Added campaign dropdown to stats and posts Growth views
- Integrated UTM data fetching with useUtmGrowthStats hook
- Campaign values display without icons/links (they're identifiers, not domains)
- Fixed DOM nesting error by removing loading skeleton from GrowthSources
- Campaign dropdown appears selected when active

9larsons committed:
Linter

9larsons committed:
DRY up a little, rm comments

9larsons committed:
Fix build issues

9larsons committed:
Lint

cmraible reviewed (changes requested):
I think this may be the first real bug our e2e tests have caught!

And the CI workflow makes it stupidly easy to see what went wrong.

You can just copy/paste the full command from the CI workflow, and (assuming you have the `gh` CLI installed) run it locally to see the test report, then click into the trace to see why it failed. 

```
gh run download 18357523609 -n playwright-report -D /tmp/playwright-$$ && npx playwright show-report /tmp/playwright-$$/playwright-report
```

Super cool 🎉

---

I haven't looked super closely at the code just yet, but will take a closer look in the morning.

9larsons committed:
Fixed loading state

cmraible committed:
Updated growth content description when campaign type is selected

cmraible committed:
Added initial draft of e2e tests for growth tab

cmraible committed:
Removed utm-growth-tabs component

This component was 99% duplicated from utm-campaign-tabs, so it made
more sense to just use utm-campaign-tabs rather than duplicating it.

cmraible committed:
Revert "Removed utm-growth-tabs component"

This reverts commit 402589fe67044e6ed45b6bce4685c933740375b5.

cmraible committed:
Fixed campaigns dropdown styling issues in the post analytics growth tab

cmraible committed:
Added e2e test for post analytics utm attribution

cmraible committed:
Fixed linter error

cmraible committed:
Updated internal import to use @ alias

ibalosh reviewed (approved):
looks good to me, added some nitpick comments only

cmraible committed:
Addressed PR feedback

cmraible committed:
Deleted redundant content-helpers unit tests

cmraible committed:
Fixed loading states for UTM campaigns

cmraible committed:
Fixed tabs showing in empty state

cmraible committed:
Fixed 'View all' footer not appearing in GrowthSources component

cmraible committed:
Fixed campaign dropdown causing invalid post_type in Growth stats

cmraible committed:
Consolidated campaign tabs components to eliminate duplication

ibalosh commented on apps/posts/src/views/PostAnalytics/Growth/components/GrowthSources.tsx:
```diff
@@ -163,33 +228,60 @@ export const GrowthSources: React.FC<SourcesCardProps> = ({
+                                </EmptyIndicator>
+                            </div>
+                        )}
+                    </>
```
agreed, duplications like in EmptyIndicator, or all the conditionals we use is something we should cleanup at some point. Since we are tight on schedule, probably big cleanup is something we should consider in future

peterzimon committed:
Simplified types in utm-campaign-tabs component

peterzimon commented on apps/shade/src/components/features/utm-growth-tabs/utm-growth-tabs.tsx:
```diff
@@ -0,0 +1,81 @@
+export const GROWTH_CAMPAIGN_OPTIONS = GROWTH_CAMPAIGN_TYPES.map(type => ({
+    value: type,
+    label: type
+}));
```
@cmraible — I updated this so it's slightly easier to read and reason, also it uses less type definition. https://github.com/TryGhost/Ghost/pull/25104/commits/09fae05cf7c6a6d04a89f0c3f5d8a6787e804994

Do you think this is better?
